schema/deploy/shipping/views.sql

```diff
@@ -740,24 +754,316 @@ comment on view shipping.return_results_v2 is
     'Version 2 of view of barcodes and presence/absence results for return of results on website';
 
 
+create or replace view shipping.fhir_encounter_details_v2 as
+
+    with
+        symptoms as (
+          select encounter_id,
+                 array_agg(distinct condition."id" order by condition."id") as symptoms,
+                 -- In our FHIR etl we give all symptoms the same onsetDateTime
+                 "onsetDateTime" as symptom_onset
+            from warehouse.encounter,
+                 jsonb_to_recordset(details -> 'Condition') as condition("id" text, "onsetDateTime" text)
+          group by encounter_id, symptom_onset
+        ),
+
+        -- This creates a loooong table of encounter_id, linkId, and all ingested "value*" fields
+        -- The following CTEs filter by linkId and uses the appropriate "value*" field to
+        -- select answer values.
+        questionnaire_responses as (
+          select encounter_id,
+                 "linkId",
+                 array_remove(array_agg("valueString" order by "valueString"), null) as string_response,
+                 -- All boolean values must be true to be considered True
+                 -- I don't think we will ever get two answers for one boolean question, but just in case.
+                 -- Jover, 18 March 2020
+                 bool_and("valueBoolean") as boolean_response,
+                 array_remove(array_agg("valueDate" order by "valueDate"), null) as date_response,
+                 array_remove(array_agg("code" order by "code"), null) as code_response
+            from warehouse.encounter,
+                 jsonb_to_recordset(details -> 'QuestionnaireResponse') as q("item" jsonb),
+                 jsonb_to_recordset("item") as response("linkId" text, "answer" jsonb),
+                 jsonb_to_recordset("answer") as answer("valueString" text, "valueBoolean" bool, "valueDate" text, "valueCoding" jsonb),
+                 jsonb_to_record("valueCoding") as code("code" text)
+          where "linkId" in ('redcap_event_name',
+                             'vaccine',
+                             'race',
+                             'insurance',
+                             'ethnicity',
+                             'travel_countries',
+                             'travel_countries_phs',
+                             'country',
```

Comment:
This `linkId` isn't used below. Should it be?

Reply:
Yes! Thank you, I missed `country` and `state` 😑